keps/sig-aws/20181127-aws-ebs-csi-driver.md

```diff
@@ -0,0 +1,73 @@
+---
+title: aws-ebs-csi-driver
+authors:
+  - "@leakingtapan"
+owning-sig: sig-aws
+reviewers:
+  - "@d-nishi"
+  - "@jsafrane"
+approvers:
+  - "@d-nishi"
+  - "@jsafrane"
+editor: TBD
+creation-date: 2018-11-27
+last-updated: 2018-11-27
+status: provisional
+---
+
+# AWS Elastic Block Store (EBS) CSI Driver
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories](#user-stories)
+        * [Static Provisioning](#static-provisioning)
+        * [Volume Schduling](#volume-scheduling)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+
+## Summary
+AWS EBS CSI Driver implements [Container Storage Interface](https://github.com/container-storage-interface/spec/tree/master) which is the standard of storage interface for container. It provides the same in-tree AWS EBS plugin features including volume creation, volume attachment, volume mounting and volume scheduling. It is also configurable on what is the EBS volume type to create, what is the file system file should be formatted, which KMS key to use to create encrypted volume, etc.
+
+## Motivation
+Similar to CNI plugins, AWS EBS CSI driver will be a stand alone plugin that lives out-of-tree of kuberenetes. Being out-of-tree, it will be benefit from being modularized, maintained and optimized without affecting kubernetes core code base. Aside from those benefits, it could also be consumed by other container orchestrators such as ECS.
+
+### Goals
+AWS EBS CSI driver will provide similar user experience as in-tree EBS plugin:
+* As an application developer, he will not even notice any difference between EBS CSI driver and in-tree plugin. His workflow will stay the same as current.
+* As an infrastructure operator, he just need to create/update storage class to use CSI driver to manage underlying storage backend.
+
+List of driver features include volume creation/deletion, volume attach/detach, volume mount/unmount, volume scheduling, create volume configurations, volume snapshotting, mount options, raw block volume, etc.
+
+### Non-Goals
+* Supporting non AWS block storage
+* Supporting other AWS storage serivces such as Dynamodb, S3, etc.
+
+## Proposal
+
+### User Stories
+
+#### Static Provisioning
+Operator creates a pre-created EBS volume on AWS and a PV that refer the EBS volume on cluster. Developer creates PVC and a Pod that uses the PVC. Then developer deploys the Pod during which time the PV will be attached to container inside Pod after PVC bonds to PV successfully.
+
+#### Volume Scheduling
```

Comment:
Add chapter about dynamic provisioning. It should be simple, however, it should mention that a different storage class is needed.